packages/alpinejs/src/directives/x-html.js

```diff
@@ -2,18 +2,22 @@ import { directive } from '../directives'
 import { initTree } from '../lifecycle'
 import { mutateDom } from '../mutation'
 
-directive('html', (el, { expression }, { effect, evaluateLater }) => {
+directive('html', (el, { expression, modifiers }, { effect, evaluateLater }) => {
     let evaluate = evaluateLater(expression)
 
     effect(() => {
         evaluate(value => {
-            mutateDom(() => {
-                el.innerHTML = value
+            value = value ?? null;
 
-                el._x_ignoreSelf = true
-                initTree(el)
-                delete el._x_ignoreSelf
-            })
+            if (!modifiers.includes('ignore-nullish') || value !== null) {
```

Comment:
```suggestion
            if (!modifiers.includes('ignore-nullish') || value != null) {
```

NIT: Then you can remove the above `value = value ?? null`

Reply:
I would rather explicitly coerce to `null` and be able to do a strict check. It makes it more clear what the intent is here.